apps/admin-gui/src/app/shared/components/roles-list/roles-page.component.html

```diff
@@ -10,344 +10,111 @@ <h1 class="page-subtitle d-flex">{{'ROLES.TITLE'| translate}}</h1>
         matTooltipPosition="left"
         [matTooltipDisabled]="selection.selected.length === 0 || !disableRemove">
         <button
-          (click)="openConfirmDialog('SPONSOR')"
-          [disabled]="selection.selected.length === 0 || disableRemove"
+          (click)="openConfirmDialog(role)"
+          [disabled]="role.primaryObject && (selection.selected.length === 0 || disableRemove)"
           mat-flat-button
           color="warn"
           class="mt-2">
           {{'ROLES.REMOVE'| translate}}
         </button>
       </span>
-      <perun-web-apps-vos-list
-        [displayedColumns]="['checkbox', 'id', 'name', 'shortName']"
-        [vos]="vos"
-        [selection]="selection">
-      </perun-web-apps-vos-list>
-    </div>
-  </mat-expansion-panel>
-
-  <mat-expansion-panel *ngIf="roles.has('PERUNADMIN')">
-    <mat-expansion-panel-header>
-      <mat-panel-title>
-        <mat-icon class="mr-2" mat-card-avatar svgIcon="perun-user-dark"></mat-icon>
-        <p class="mt-auto mb-auto">{{'ROLES.PERUN_ADMIN' | translate}}</p>
-      </mat-panel-title>
-    </mat-expansion-panel-header>
-    <button
-      (click)="openConfirmDialog('PERUNADMIN', false)"
-      mat-flat-button
-      color="warn"
-      class="mt-2">
-      {{'ROLES.REMOVE'| translate}}
-    </button>
-  </mat-expansion-panel>
-
-  <mat-expansion-panel *ngIf="roles.has('PERUNOBSERVER')">
-    <mat-expansion-panel-header>
-      <mat-panel-title>
-        <mat-icon class="mr-2" mat-card-avatar svgIcon="perun-user-dark"></mat-icon>
-        <p class="mt-auto mb-auto">{{'ROLES.PERUN_OBSERVER' | translate}}</p>
-      </mat-panel-title>
-    </mat-expansion-panel-header>
-    <button
-      (click)="openConfirmDialog('PERUNOBSERVER', false)"
-      mat-flat-button
-      color="warn"
-      class="mt-2">
-      {{'ROLES.REMOVE'| translate}}
-    </button>
-  </mat-expansion-panel>
-
-  <mat-expansion-panel (opened)="getInnerKeys(role)" *ngFor="let role of roleNames">
-    <mat-expansion-panel-header>
-      <mat-panel-title>
-        <mat-icon class="mr-2" mat-card-avatar svgIcon="perun-user-dark"></mat-icon>
-        <p class="mt-auto mb-auto">{{role | translate}}</p>
-      </mat-panel-title>
-    </mat-expansion-panel-header>
-
-    <p *ngFor="let key of currentIds" class="role-content">{{key[0]}} ids: {{key[1]}}</p>
+      <div *ngIf="panel.expanded" class="position-relative">
+        <div *perunWebAppsLoader="loading; indicator: spinner">
+          <div *ngIf="role.primaryObject === 'Vo' || role.roleName === 'MEMBERSHIP'" class="mb-3">
+            <div *ngIf="showDescription">
+              {{('ROLES.' + role.roleName + '_VOS_' + entityType) | translate}}
+            </div>
+            <perun-web-apps-vos-list
+              [displayedColumns]="role.roleName === 'MEMBERSHIP' ? ['id', 'name', 'shortName'] : ['checkbox', 'id', 'name', 'shortName']"
+              [vos]="vos | async"
+              [selection]="selection">
```

Comment:
Yes, the annoying thing is that unlike most perun objects the "richer" facility does not extend facility and therefore does not extend perun bean. Right now the only solution I see is to have 2 selection models, but I would rather have a compiler error. The error is harmless I would say, since we don't mix these 2 different types at any point.